src/Common/MemoryTracker.cpp

```diff
@@ -294,33 +293,18 @@ AllocationTrace MemoryTracker::allocImpl(Int64 size, bool throw_if_memory_exceed
         }
     }
 
-    Int64 limit_to_check = current_hard_limit;
-
 #if USE_JEMALLOC
-    if (level == VariableContext::Global && allow_use_jemalloc_memory.load(std::memory_order_relaxed))
+    if (level == VariableContext::Global && will_be > soft_limit.load(std::memory_order_relaxed)
```

Comment:
we don't set soft limit for global tracker so maybe use hard_limit here or introduce another setting